### ISSUE TITLE
Remove pw-selector.js injection, use custom locator.ts for recording JS locators

### ISSUE THREAD

**Issue by stevez** [stevez/playwright-repl#308]
## Context

Follow-up to #305 (Option 1).

`pw-selector.js` injects Playwright's `generateSelectorSimple()` + `asLocator()` into the utility world to tag elements with `data-pw-locator` during recording. However, this produces the same verbose chained locators as Playwright's codegen:

```js
await page.locator('div').filter({ hasText: /^npmyarnpnpm.../ }).getByRole('tab').first().click();
```

Our custom `locator.ts` already produces cleaner results:

```js
await page.getByRole('tab', { name: 'npm' }).click();
```

## Changes

- Remove `data-pw-locator` priority check in `generateLocator()` (`locator.ts`)
- Remove `installPwSelectors()` and related state in `background.ts`
- Delete `pw-selector.js`
- Remove `*.js?raw` type declaration in `types.d.ts`
- Remove associated tests (`pw-selector-source.test.ts`, locator test for `data-pw-locator`)
- Keep the `await app.detach()` fix from #296

## Related

- #305 — parent investigation issue
- #296 — introduced `pw-selector.js` (this reverts the locator portion)
- #305 Option 2 (CDP `_generateLocatorString()`) can be layered on later if needed
